kotlinlib/src/mill/kotlinlib/js/KotlinJsModule.scala

```diff
@@ -105,54 +107,68 @@ trait KotlinJsModule extends KotlinModule { outer =>
     Task.Command { run(args)() }
 
   override def run(args: Task[Args] = Task.Anon(Args())): Command[Unit] = Task.Command {
-    val binaryKind = kotlinJSBinaryKind()
-    if (binaryKind.isEmpty || binaryKind.get != BinaryKind.Executable) {
-      T.log.error("Run action is only allowed for the executable binary")
+    runJsBinary(
+      args = args(),
+      binaryKind = kotlinJSBinaryKind(),
+      moduleKind = moduleKind(),
+      binaryDir = linkBinary().classes.path,
+      runTarget = kotlinJSRunTarget(),
+      envArgs = T.env,
+      workingDir = T.dest
+    ).map(_ => ()).getOrThrow
+  }
+
+  override def runMainLocal(
+      @arg(positional = true) mainClass: String,
+      args: String*
+  ): Command[Unit] = Task.Command[Unit] {
+    mill.api.Result.Failure("runMain is not supported in Kotlin/JS.")
+  }
+
+  override def runMain(@arg(positional = true) mainClass: String, args: String*): Command[Unit] =
+    Task.Command[Unit] {
+      mill.api.Result.Failure("runMain is not supported in Kotlin/JS.")
     }
 
-    val moduleKind = this.moduleKind()
+  protected[js] def runJsBinary(
```

Comment:
I'd prefer the name `runKotlinJsBinary`, since chances are high the user project may also have some Scala.JS or other JS modules.